scripts/helpers.py

```diff
@@ -565,3 +566,27 @@ def locate_bus(
         return gdf_co[gdf_co.geometry == min(gdf_co.geometry, key=(point.distance))][
             col
         ].item()  # looks for closest one shape=node
+
+
+def get_last_commit_message(path):
+    """
+    Function to get the last PyPSA-Earth Git commit message
+    Returns
+    -------
+    result : string
+    """
+    _logger = logging.getLogger(__name__)
+    try:
+        last_commit_message = (
+            subprocess.check_output(
+                ["git", "log", "-n", "1", "--pretty=format:%H %s"],
+                cwd=path,
+                stderr=subprocess.STDOUT,
+            )
+            .decode()
+            .strip()
+        )
+        return last_commit_message
+    except subprocess.CalledProcessError as e:
+        _logger.warning(f"Error executing Git: {e}")
+        return None
```

Comment:
According to [the documentation](https://docs.python.org/3/library/subprocess.html), it seems that the cwd option overrides the cwd.

It may be a good idea to create a backup of cwd and restore it after the execution in an error-proven way.

To do this, for example, we may:
1. before the try, initialize the last_commit_message to None and store cwd in a local variable
2. in the try/except, drop the return statements
3. after the try/except, restore the original cwd path
4. return the last_commit_message

This may be worth trying to check if it solves the problem.
It may be good to reflect these changes into pypsa-earth as well, once validated.

Reply:
Hi @davide-f ,
thanks for the solution approach. Sadly it seems not to work. 
Do you have any other ideas? Im pretty much empty with solution ideas.